scripts/tests/test_zkai_attention_kv_ezkl_same_surface_export_probe.py

```
@@ -0,0 +1,123 @@
+        with self.assertRaisesRegex(PROBE.SameSurfaceExportProbeError, "attention_outputs width drift"):
+            PROBE.source_shape(source, PROBE.DEFAULT_SOURCE)
+
+    def test_validation_rejects_candidate_overclaim(self) -> None:
+        payload = PROBE.build_probe()
+        payload["candidate_adapters"][0]["proof_generated"] = True
+        payload["candidate_matrix_commitment"] = PROBE.blake2b_commitment(
+            payload["candidate_adapters"],
+            "ptvm:zkai:attention-kv-ezkl-same-surface-candidates:v1",
+        )
+
+        with self.assertRaisesRegex(PROBE.SameSurfaceExportProbeError, "candidate adapter matrix drift"):
```

Ruff (0.15.13)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)